### PR TITLE
feat(feature_matrix): impute values for gene attribute cols

### PR DESCRIPTION
**Pull request by ireneisdoomed**
The `GeneCountFeature`, `ProteinGeneCountFeature`, and `ProteinCodingFeature` feature factories don't generate annotation for all pairs of study locus/genes.

There are some pairs that are annotated with genes that come from colocalisation, and therefore generate nulls in the features that refer to gene attributes when we convert to the wide format feature matrix.

This is not correct, since we shouldn't have any null information for them. To avoid the null values, we implement an imputation strategy using the average value for a gene. Using the mean is a sensible approach, given that all 3 columns are fairly symmetrical.

## ✨ Context
This PR closes https://github.com/opentargets/issues/issues/3602

## 🛠 What does this PR implement
-  `L2GFeautreMatrix.fill_na` now has an imputation strategy for the mentioned columns
- Semantic test

## 🚦 Before submitting

- [x] Do these changes cover one single feature (one change at a time)?
- [x] Did you read the [contributor guideline](https://opentargets.github.io/gentropy/development/contributing/#contributing-checklist)?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you make sure there is no commented out code in this PR?
- [x] Did you follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standards in PR title and commit messages?
- [x] Did you make sure the branch is up-to-date with the `dev` branch?
- [x] Did you write any new necessary tests?
- [x] Did you make sure the changes pass local tests (`make test`)?
- [x] Did you make sure the changes pass pre-commit rules (e.g `poetry run pre-commit run --all-files`)?

### CODE REVIEW

File: src/gentropy/dataset/l2g_feature_matrix.py [hunk 1]
```diff
@@ -128,18 +130,32 @@ def calculate_feature_missingness_rate(
...
+        cols_to_impute = ["proteinGeneCount500kb", "geneCount500kb", "isProteinCoding"]
+        for col in cols_to_impute:
+            if col not in self._df.columns:
+                continue
+            else:
+                self._df = self._df.withColumn(
+                    col,
+                    f.when(
+                        f.col(col).isNull(),
+                        f.mean(f.col(col)).over(Window.partitionBy("geneId")),
```
Comment (addramir) [inline src/gentropy/dataset/l2g_feature_matrix.py:155]: Why the window is by geneId? Shouldn't it be by studyLocusId?

Comment (ireneisdoomed) [reply src/gentropy/dataset/l2g_feature_matrix.py:155]: No. The studyLocus/gene ID pair will only have a null.

**Review by addramir — APPROVED** [review]

File: src/gentropy/dataset/l2g_feature_matrix.py [hunk 2]
```diff
@@ -128,18 +130,32 @@ def calculate_feature_missingness_rate(
...
         Args:
-            value (float): Value to replace missing values with. Defaults to 0.0.
+            na_value (float): Value to replace missing values with. Defaults to 0.0.
             subset (list[str] | None): Subset of columns to consider. Defaults to None.
 
         Returns:
             L2GFeatureMatrix: L2G feature matrix dataset
         """
-        self._df = self._df.fillna(value, subset=subset)
+        cols_to_impute = ["proteinGeneCount500kb", "geneCount500kb", "isProteinCoding"]
```
Comment (addramir) [inline src/gentropy/dataset/l2g_feature_matrix.py:146]: I realised:
it should be 
credibleSetConfidence instead of isProteinCoding

**Comment by ireneisdoomed** [comment]
No more nulls for these features
```
spark.read.parquet("gs://ot-team/irene/l2g/05112024/feature_matrix")
.filter((f.col("isProteinCoding").isNull()) | (f.col("proteinGeneCount500kb").isNull()) | (f.col("geneCount500kb").isNull())).count()
0
```

Stats before imputing:
```
+-------+---------------------+------------------+-------------------+          
|summary|proteinGeneCount500kb|geneCount500kb    |isProteinCoding    |
+-------+---------------------+------------------+-------------------+
|count  |21958279             |51419883          |96650848           |
|mean   |14.132147104971205   |26.522517311056504|0.41342734002706316|
|stddev |8.814968775828772    |14.077634970094412|0.49244814656397656|
|min    |1.0                  |1.0               |0.0                |
|max    |50.0                 |123.0             |1.0                |
+-------+---------------------+------------------+-------------------+
```

Stats after imputing:
```
+-------+---------------------+------------------+-------------------+          
|summary|proteinGeneCount500kb|geneCount500kb    |isProteinCoding    |
+-------+---------------------+------------------+-------------------+
|count  |108478020            |108478020         |108478020          |
|mean   |11.225733489604622   |25.033421802868453|0.41418731970020634|
|stddev |8.284680617280541    |13.765179858074502|0.46804225689279627|
|min    |0.0                  |0.0               |0.0                |
|max    |50.0                 |123.0             |1.0                |
+-------+---------------------+------------------+-------------------+
```

No drastic changes in the distribution after the imputation.

**Pull request merged**